src/components/sign/Sign.js

```diff
@@ -20,10 +20,18 @@ class Sign extends Component {
     handleDeny = e => {
         e.preventDefault();
         Mixpanel.track("SIGN Deny the transaction")
-        const { callbackUrl, meta } = this.props;
+        const { callbackUrl, meta, actionStatus } = this.props;
         // TODO: Dispatch action for app redirect?
         if (this.props.callbackUrl) {
-            window.location.href = addQueryParams(callbackUrl, { meta, errorCode: 'userRejected' })
+            let errorMessage = 'userRejected';
+            const signTxStatus = actionStatus.SIGN_AND_SEND_TRANSACTIONS;
+            if (signTxStatus?.success === false) {
+                errorMessage = 'Unknown error occured';
+                if (signTxStatus.errorMessage) {
+                    errorMessage = signTxStatus.errorMessage;
```

Comment:
is there any way we can pass through `type` from `near-api-js` https://github.com/near/near-api-js/blob/ee975a1edd84dff955302453fbcca1413d362bb7/src/utils/errors.ts#L14? (when it's available)

Reply:
@vgrichina good idea! I added `errorType`. Please check again.